lib/util/http/rate-limits.ts

```diff
@@ -0,0 +1,87 @@
+import is from '@sindresorhus/is';
+import { matchesHost } from '../host-rules';
+import * as hostRules from '../host-rules';
+import type { RateLimitRule } from './types';
+
+const defaults: RateLimitRule[] = [
+  {
+    // https://guides.rubygems.org/rubygems-org-rate-limits/
+    matchHost: 'rubygems.org',
+    throttleMs: 125,
+  },
+  {
+    // https://crates.io/data-access#api
+    matchHost: 'https://crates.io/api/',
+    throttleMs: 1000,
+  },
+  {
+    matchHost: '*',
+    concurrency: 16,
```

Comment:
@rarkins do we really want this? 🤔

Reply:
Probably a good idea. For example npm v9 had a default of 15 concurrent requests, which was lost in v10, and apparently it caused some hosts load problems.